### PR TITLE
Implement i8042 diagnostic dump

### PR DESCRIPTION
**Pull request by FeralChild64**
Implemented i8042 diagnostic dump based on logs from real chip communication found on Vogons forum.

### CODE REVIEW

File: src/hardware/input/intel8042.cpp [hunk 1]
```diff
@@ -892,6 +913,17 @@ static uint8_t read_data_port(io_port_t, io_width_t) // port 0x60
 		return data_byte;
 	}
 
+	if (is_diagnostic_dump && !buffer_num_used) {
+		// Diagnostic dump finished
+		is_diagnostic_dump = false;
```
Comment (kcgen) [inline src/hardware/input/intel8042.cpp:918]: If the program doesn't read or drain the buffer, is there a chance we can get trapped in `is_diagnostic_dump`-mode?

Comment (FeralChild64) [reply src/hardware/input/intel8042.cpp:918]: OK, should be correct now.

**Review by kcgen — APPROVED** [review]
:+1: , looks good, @FeralChild64 !

**Comment by kcgen** [comment]
(Feel free to merge when it's ready)

**Pull request merged**